storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -97,56 +97,108 @@ func (s *commitLogSource) Read(
 
 	var (
 		namespace   = ns.ID()
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
+		numShards   = s.findHighestShard(shardsTimeRanges) + 1
+		numConc     = s.opts.EncodingConcurrency()
 		bopts       = s.opts.ResultOptions()
 		blopts      = bopts.DatabaseBlockOptions()
 		blockSize   = ns.Options().RetentionOptions().BlockSize()
 		encoderPool = bopts.DatabaseBlockOptions().EncoderPool()
-		errs        = 0
+		workerErrs  = make([]int, numConc)
 	)
-	for iter.Next() {
-		series, dp, unit, annotation := iter.Current()
 
-		// check if the series belongs to current namespace being bootstrapped
-		if !namespace.Equal(series.Namespace) {
-			continue
+	// +1 so we can use the shard number as an index throughout without constantly
```

Comment:
nit: move comment adjacent to numShards (better) or delete

Reply:
fixed